### DISCUSSION TITLE
gitea 容器内存放的数据在哪？

### DISCUSSION THREAD

**Discussion by liminngs** [OpenCSGs/csghub#624] [Q&A]
gitea 容器内存放的数据在哪？

**Comment by Rader** [comment]
默认是在docker compose启动服务的当前目录下。如果不确定的话，可以通过` docker inspect` 查看 gitea对应container 的 `Mounts` 信息，例如：
```
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/Users/leida/projects/opencsg/csghub-server/local_only/gitea",
                "Destination": "/bitnami/gitea",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
```

**Comment by liminngs** [comment]
> 默认是在docker compose启动服务的当前目录下。如果不确定的话，可以通过` docker inspect` 查看 gitea对应container 的 `Mounts` 信息，例如：
> 
> ```
>         "Mounts": [
>             {
>                 "Type": "bind",
>                 "Source": "/Users/leida/projects/opencsg/csghub-server/local_only/gitea",
>                 "Destination": "/bitnami/gitea",
>                 "Mode": "rw",
>                 "RW": true,
>                 "Propagation": "rprivate"
>             }
>         ],
> ```

"Mounts": [],
没有挂着信息  该存容器内哪个目录下进挂载？

**Comment by Rader** [comment]
docker-compose 文件应该有挂载。您不是通过 docker compose启动的吗？
```
    volumes:
      - ./gitea:/bitnami/gitea
    networks:
```

**Comment by wayneliu0019** [comment]
补充：

如果您是通过`https://github.com/OpenCSGs/CSGHub/blob/main/all-in-one-CN.yml` 这里的all in one脚本部署启动的，那么gitea服务是没有把数据目录挂载到宿主机磁盘的，因为这种方式是fast try，没有做持久化，数据，配置都保存在容器内，重启后就会丢失；

使用`https://github.com/OpenCSGs/CSGHub/tree/main/script/all_in_one` 这个目录下的docker-compose.ym启动的csghub服务可以作为准生产环境使用，是做了持久化的，如下
```
.....
    networks:
      - opencsg
    privileged: true
    restart: always
    volumes:
      - ./gitdata:/bitnami/gitea
      - ./gitlog:/bitnami/logs
```
可以看到gitea数据目录是保存在宿主机上的，这些主要是repo数据，其他数据比如gitea用户，key，权限等信息是保存在db中的。
